app/src/main/java/ua/dancecamp/vmedvediev/ua21summerdancecamp/UI/login/LoginActivity.kt

```diff
@@ -0,0 +1,143 @@
+package ua.dancecamp.vmedvediev.ua21summerdancecamp.UI.login
+
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import android.support.v7.app.AppCompatActivity
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import kotlinx.android.synthetic.main.activity_login.*
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.*
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.UI.Router
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.authentication.EncryptionServices
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.mappers.RealmCredentialsMapper
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.services.SystemServices
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.mappers.RealmEventMapper
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.mappers.RealmSettingsMapper
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.model.Credentials
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.repository.Repository
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.services.openSecuritySettings
+
+class LoginActivity : AppCompatActivity() {
+
+    companion object {
+        private const val CREDENTIALS_ID = "1"
+    }
+
+    private val loginViewModel by lazy {
+        ViewModelProviders.of(this, LoginViewModel(Repository(RealmEventMapper(),
+                RealmSettingsMapper(), RealmCredentialsMapper())).LoginViewModelFactory())
+                .get(LoginViewModel::class.java)
+    }
+    private val systemServices by lazy(LazyThreadSafetyMode.NONE) { SystemServices(this) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+
+        if (systemServices.isFingerprintHardwareAvailable()) {
+            allowFingerprintCheckBox.visibility = View.VISIBLE
+        }
+        allowFingerprintCheckBox.setOnCheckedChangeListener { _, checked -> onAllowFingerprint(checked) }
```

Comment:
No need to execute logic connected with fingerprint if fingerprint is not available